### PR TITLE
ELY-2271 IdentityCredentials.verify(Supplier<Provider[]>, evidence, h…

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ELY-2271
https://issues.redhat.com/browse/ELY-2272

Please note the deprecations I've added.

### CODE REVIEW

File: auth/realm/base/src/main/java/org/wildfly/security/auth/realm/CachingModifiableSecurityRealm.java [hunk 1]
```diff
@@ -53,6 +55,17 @@ public CachingModifiableSecurityRealm(CacheableSecurityRealm realm, RealmIdentit
         super(realm, cache);
     }
 
+    /**
+     * Creates a new instance.
+     *
+     * @param realm the {@link CacheableSecurityRealm} whose {@link RealmIdentity} should be cached..
```
Comment (fjuma) [inline auth/realm/base/src/main/java/org/wildfly/security/auth/realm/CachingModifiableSecurityRealm.java:61]: Just a small comment, there's an extra period at the end of this line

File: http/basic/src/main/java/org/wildfly/security/http/basic/BasicMechanismFactory.java [hunk 2]
```diff
@@ -45,7 +45,7 @@
     public BasicMechanismFactory() {
     }
 
-    public BasicMechanismFactory(final Provider provider) {
+    public BasicMechanismFactory(final Provider... providers) {
```
Comment (fjuma) [inline http/basic/src/main/java/org/wildfly/security/http/basic/BasicMechanismFactory.java:48]: Just FYI for other reviewers, this isn't public API.

**Review by fjuma — APPROVED** [review]

**Comment by fjuma** [comment]
@tristantarrant This looks good to me. Just noticed, this PR was submitted against the 1.x branch. If you'd like to get this fix into 1.17.x, you can update this to be directly against that branch. We then forward port to 1.x as part of our release process.

File: auth/realm/base/src/main/java/org/wildfly/security/auth/realm/CachingModifiableSecurityRealm.java (hunk 1 above)
Comment (tristantarrant) [reply auth/realm/base/src/main/java/org/wildfly/security/auth/realm/CachingModifiableSecurityRealm.java]: FIxed this one and the one above

**Comment by fjuma** [comment]
Thanks, @tristantarrant! I've now created the 1.17.x branch. If you could update this PR to be based off that branch that would be great. (I tried changing the base branch myself but more work is needed.)

**Comment by tristantarrant** [comment]
1.17 backport: https://github.com/wildfly-security/wildfly-elytron/pull/1641

**Comment by fjuma** [comment]
Perfect, thanks @tristantarrant! I'm going to close this one now in favour of https://github.com/wildfly-security/wildfly-elytron/pull/1641. We forward port the changes from maintenance branches to 1.x as part of our release process so we don't need the additional PR.

**Pull request closed without merging**